src/main/scala/com/microsoft/hyperspace/telemetry/HyperspaceEvent.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.telemetry
+
+import org.apache.commons.lang.StringUtils
+
+import com.microsoft.hyperspace.index.IndexLogEntry
+
+/**
+ * Base class for all Hyperspace events.
+ */
+trait HyperspaceEvent
+
+/**
+ * General index CRUD event.
+ *
+ * @param sparkUser user of spark application.
+ * @param appId spark application id.
+ * @param appName spark application name.
+ * @param index related indexes.
+ * @param message message about event.
+ */
+class HyperspaceIndexCRUDEvent(
+    sparkUser: String,
+    appId: String,
+    appName: String,
+    index: IndexLogEntry,
+    message: String = StringUtils.EMPTY)
+    extends HyperspaceEvent
+
+case class CreateActionEvent(
```

Comment:
do you want to add a comment for each event?

Reply:
thanks @imback82 , done